warning: the feature `specialization` is incomplete and may not be safe to use and/or cause compiler crashes
 --> $DIR/lib.rs:1:12
  |
1 | #![feature(specialization)]
  |            ^^^^^^^^^^^^^^
  |
  = note: `#[warn(incomplete_features)]` on by default
  = note: see issue #31844 <https://github.com/rust-lang/rust/issues/31844> for more information

warning: the feature `specialization` is incomplete and may not be safe to use and/or cause compiler crashes
 --> $DIR/lib.rs:1:12
  |
1 | #![feature(specialization)]
  |            ^^^^^^^^^^^^^^
  |
  = note: `#[warn(incomplete_features)]` on by default
  = note: see issue #31844 <https://github.com/rust-lang/rust/issues/31844> for more information

warning: trait objects without an explicit `dyn` are deprecated
   --> $DIR/uri_display.rs:299:43
    |
299 | impl<'a, P: UriPart> fmt::Display for &'a UriDisplay<P> {
    |                                           ^^^^^^^^^^^^^ help: use `dyn`: `dyn UriDisplay<P>`
    |
    = note: `#[warn(bare_trait_objects)]` on by default

warning: trait objects without an explicit `dyn` are deprecated
   --> $DIR/uri_display.rs:299:43
    |
299 | impl<'a, P: UriPart> fmt::Display for &'a UriDisplay<P> {
    |                                           ^^^^^^^^^^^^^ help: use `dyn`: `dyn UriDisplay<P>`
    |
    = note: `#[warn(bare_trait_objects)]` on by default

warning: `...` range patterns are deprecated
 --> $DIR/checkers.rs:9:12
  |
9 |         '0'...'9' | 'A'...'Z' | '^'...'~' | '#'...'\''
  |            ^^^ help: use `..=` for an inclusive range
  |
  = note: `#[warn(ellipsis_inclusive_range_patterns)]` on by default

warning: `...` range patterns are deprecated
 --> $DIR/checkers.rs:9:12
  |
9 |         '0'...'9' | 'A'...'Z' | '^'...'~' | '#'...'\''
  |            ^^^ help: use `..=` for an inclusive range
  |
  = note: `#[warn(ellipsis_inclusive_range_patterns)]` on by default

warning: `...` range patterns are deprecated
 --> $DIR/checkers.rs:9:24
  |
9 |         '0'...'9' | 'A'...'Z' | '^'...'~' | '#'...'\''
  |                        ^^^ help: use `..=` for an inclusive range

warning: `...` range patterns are deprecated
 --> $DIR/checkers.rs:9:24
  |
9 |         '0'...'9' | 'A'...'Z' | '^'...'~' | '#'...'\''
  |                        ^^^ help: use `..=` for an inclusive range

warning: `...` range patterns are deprecated
 --> $DIR/checkers.rs:9:36
  |
9 |         '0'...'9' | 'A'...'Z' | '^'...'~' | '#'...'\''
  |                                    ^^^ help: use `..=` for an inclusive range

warning: `...` range patterns are deprecated
 --> $DIR/checkers.rs:9:36
  |
9 |         '0'...'9' | 'A'...'Z' | '^'...'~' | '#'...'\''
  |                                    ^^^ help: use `..=` for an inclusive range

warning: `...` range patterns are deprecated
 --> $DIR/checkers.rs:9:48
  |
9 |         '0'...'9' | 'A'...'Z' | '^'...'~' | '#'...'\''
  |                                                ^^^ help: use `..=` for an inclusive range

warning: 6 warnings emitted

warning: `...` range patterns are deprecated
 --> $DIR/checkers.rs:9:48
  |
9 |         '0'...'9' | 'A'...'Z' | '^'...'~' | '#'...'\''
  |                                                ^^^ help: use `..=` for an inclusive range

warning: 6 warnings emitted

warning: the feature `specialization` is incomplete and may not be safe to use and/or cause compiler crashes
 --> $DIR/lib.rs:1:12
  |
1 | #![feature(specialization)]
  |            ^^^^^^^^^^^^^^
  |
  = note: `#[warn(incomplete_features)]` on by default
  = note: see issue #31844 <https://github.com/rust-lang/rust/issues/31844> for more information

warning: trait objects without an explicit `dyn` are deprecated
   --> $DIR/response.rs:563:27
    |
563 |     body: Option<Body<Box<io::Read + 'r>>>,
    |                           ^^^^^^^^^^^^^ help: use `dyn`: `dyn io::Read + 'r`
    |
    = note: `#[warn(bare_trait_objects)]` on by default

warning: trait objects without an explicit `dyn` are deprecated
   --> $DIR/response.rs:892:48
    |
892 |     pub fn body(&mut self) -> Option<Body<&mut io::Read>> {
    |                                                ^^^^^^^^ help: use `dyn`: `dyn io::Read`

warning: trait objects without an explicit `dyn` are deprecated
   --> $DIR/response.rs:969:52
    |
969 |     pub fn take_body(&mut self) -> Option<Body<Box<io::Read + 'r>>> {
    |                                                    ^^^^^^^^^^^^^ help: use `dyn`: `dyn io::Read + 'r`

warning: `...` range patterns are deprecated
  --> $DIR/toml_ext.rs:27:12
   |
27 |         '0'...'9' | 'A'...'Z' | 'a'...'z' | '_' | '-' => true,
   |            ^^^ help: use `..=` for an inclusive range
   |
   = note: `#[warn(ellipsis_inclusive_range_patterns)]` on by default

warning: `...` range patterns are deprecated
  --> $DIR/toml_ext.rs:27:24
   |
27 |         '0'...'9' | 'A'...'Z' | 'a'...'z' | '_' | '-' => true,
   |                        ^^^ help: use `..=` for an inclusive range

warning: `...` range patterns are deprecated
  --> $DIR/toml_ext.rs:27:36
   |
27 |         '0'...'9' | 'A'...'Z' | 'a'...'z' | '_' | '-' => true,
   |                                    ^^^ help: use `..=` for an inclusive range

warning: trait objects without an explicit `dyn` are deprecated
  --> $DIR/data.rs:18:63
   |
18 |     self::HttpReader<&'a mut hyper::buffer::BufReader<&'b mut NetworkStream>>;
   |                                                               ^^^^^^^^^^^^^ help: use `dyn`: `dyn NetworkStream`

warning: trait objects without an explicit `dyn` are deprecated
   --> $DIR/data.rs:108:41
    |
108 |         fn concrete_stream(stream: &mut NetworkStream) -> Option<NetStream> {
    |                                         ^^^^^^^^^^^^^ help: use `dyn`: `dyn NetworkStream`

warning: trait objects without an explicit `dyn` are deprecated
   --> $DIR/handler.rs:155:36
    |
155 |     fn clone_handler(&self) -> Box<Handler>;
    |                                    ^^^^^^^ help: use `dyn`: `dyn Handler`

warning: trait objects without an explicit `dyn` are deprecated
   --> $DIR/handler.rs:160:36
    |
160 |     fn clone_handler(&self) -> Box<Handler> {
    |                                    ^^^^^^^ help: use `dyn`: `dyn Handler`

warning: trait objects without an explicit `dyn` are deprecated
   --> $DIR/handler.rs:165:20
    |
165 | impl Clone for Box<Handler> {
    |                    ^^^^^^^ help: use `dyn`: `dyn Handler`

warning: trait objects without an explicit `dyn` are deprecated
   --> $DIR/handler.rs:167:28
    |
167 |     fn clone(&self) -> Box<Handler> {
    |                            ^^^^^^^ help: use `dyn`: `dyn Handler`

warning: trait objects without an explicit `dyn` are deprecated
 --> $DIR/fairings.rs:8:27
  |
8 |     all_fairings: Vec<Box<Fairing>>,
  |                           ^^^^^^^ help: use `dyn`: `dyn Fairing`

warning: trait objects without an explicit `dyn` are deprecated
  --> $DIR/fairings.rs:22:43
   |
22 |     pub fn attach(&mut self, fairing: Box<Fairing>, mut rocket: Rocket) -> Rocket {
   |                                           ^^^^^^^ help: use `dyn`: `dyn Fairing`

warning: trait objects without an explicit `dyn` are deprecated
  --> $DIR/fairings.rs:35:36
   |
35 |     fn add(&mut self, fairing: Box<Fairing>) {
   |                                    ^^^^^^^ help: use `dyn`: `dyn Fairing`

warning: trait objects without an explicit `dyn` are deprecated
  --> $DIR/error.rs:30:17
   |
30 |     Unknown(Box<::std::error::Error + Send + Sync>)
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use `dyn`: `dyn ::std::error::Error + Send + Sync`

warning: trait objects without an explicit `dyn` are deprecated
  --> $DIR/route.rs:22:22
   |
22 |     pub handler: Box<Handler>,
   |                      ^^^^^^^ help: use `dyn`: `dyn Handler`

warning: 18 warnings emitted

error: `has_one` route uri expects 1 parameter but 0 were supplied
  --> $DIR/typed-uris-bad-params.rs:22:10
   |
22 |     uri!(has_one); //~ ERROR expects 1 parameter but 0
   |          ^^^^^^^
   |
   = note: expected parameter: id: i32

error: `has_one` route uri expects 1 parameter but 2 were supplied
  --> $DIR/typed-uris-bad-params.rs:24:19
   |
24 |     uri!(has_one: 1, 23); //~ ERROR expects 1 parameter but 2
   |                   ^^^^^
   |
   = note: expected parameter: id: i32

error: `has_one` route uri expects 1 parameter but 2 were supplied
  --> $DIR/typed-uris-bad-params.rs:25:19
   |
25 |     uri!(has_one: "Hello", 23, ); //~ ERROR expects 1 parameter but 2
   |                   ^^^^^^^^^^^^
   |
   = note: expected parameter: id: i32

error: `has_one_guarded` route uri expects 1 parameter but 2 were supplied
  --> $DIR/typed-uris-bad-params.rs:26:27
   |
26 |     uri!(has_one_guarded: "hi", 100); //~ ERROR expects 1 parameter but 2
   |                           ^^^^^^^^^
   |
   = note: expected parameter: id: i32

error: `has_two` route uri expects 2 parameters but 3 were supplied
  --> $DIR/typed-uris-bad-params.rs:28:19
   |
28 |     uri!(has_two: 10, "hi", "there"); //~ ERROR expects 2 parameters but 3
   |                   ^^^^^^^^^^^^^^^^^
   |
   = note: expected parameters: id: i32, name: String

error: `has_two` route uri expects 2 parameters but 1 was supplied
  --> $DIR/typed-uris-bad-params.rs:29:19
   |
29 |     uri!(has_two: 10); //~ ERROR expects 2 parameters but 1
   |                   ^^
   |
   = note: expected parameters: id: i32, name: String

error: invalid parameters for `has_one` route uri
  --> $DIR/typed-uris-bad-params.rs:31:19
   |
31 |     uri!(has_one: id = 100, name = "hi"); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32
help: unknown parameter: `name`
  --> $DIR/typed-uris-bad-params.rs:31:29
   |
31 |     uri!(has_one: id = 100, name = "hi"); //~ ERROR invalid parameters
   |                             ^^^^

error: invalid parameters for `has_one` route uri
  --> $DIR/typed-uris-bad-params.rs:34:19
   |
34 |     uri!(has_one: name = 100, id = 100); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32
help: unknown parameter: `name`
  --> $DIR/typed-uris-bad-params.rs:34:19
   |
34 |     uri!(has_one: name = 100, id = 100); //~ ERROR invalid parameters
   |                   ^^^^

error: invalid parameters for `has_one` route uri
  --> $DIR/typed-uris-bad-params.rs:37:19
   |
37 |     uri!(has_one: name = 100, age = 50, id = 100); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32
help: unknown parameters: `name`, `age`
  --> $DIR/typed-uris-bad-params.rs:37:19
   |
37 |     uri!(has_one: name = 100, age = 50, id = 100); //~ ERROR invalid parameters
   |                   ^^^^        ^^^

error: invalid parameters for `has_one` route uri
  --> $DIR/typed-uris-bad-params.rs:40:19
   |
40 |     uri!(has_one: name = 100, age = 50, id = 100, id = 50); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32
help: unknown parameters: `name`, `age`
  --> $DIR/typed-uris-bad-params.rs:40:19
   |
40 |     uri!(has_one: name = 100, age = 50, id = 100, id = 50); //~ ERROR invalid parameters
   |                   ^^^^        ^^^
help: duplicate parameter: `id`
  --> $DIR/typed-uris-bad-params.rs:40:51
   |
40 |     uri!(has_one: name = 100, age = 50, id = 100, id = 50); //~ ERROR invalid parameters
   |                                                   ^^

error: invalid parameters for `has_one` route uri
  --> $DIR/typed-uris-bad-params.rs:44:19
   |
44 |     uri!(has_one: id = 100, id = 100); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32
help: duplicate parameter: `id`
  --> $DIR/typed-uris-bad-params.rs:44:29
   |
44 |     uri!(has_one: id = 100, id = 100); //~ ERROR invalid parameters
   |                             ^^

error: invalid parameters for `has_one` route uri
  --> $DIR/typed-uris-bad-params.rs:47:19
   |
47 |     uri!(has_one: id = 100, id = 100, ); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32
help: duplicate parameter: `id`
  --> $DIR/typed-uris-bad-params.rs:47:29
   |
47 |     uri!(has_one: id = 100, id = 100, ); //~ ERROR invalid parameters
   |                             ^^

error: invalid parameters for `has_one` route uri
  --> $DIR/typed-uris-bad-params.rs:50:19
   |
50 |     uri!(has_one: name = "hi"); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32
   = help: missing parameter: `id`
help: unknown parameter: `name`
  --> $DIR/typed-uris-bad-params.rs:50:19
   |
50 |     uri!(has_one: name = "hi"); //~ ERROR invalid parameters
   |                   ^^^^

error: invalid parameters for `has_one_guarded` route uri
  --> $DIR/typed-uris-bad-params.rs:54:27
   |
54 |     uri!(has_one_guarded: cookies = "hi", id = 100); //~ ERROR invalid parameters
   |                           ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32
help: unknown parameter: `cookies`
  --> $DIR/typed-uris-bad-params.rs:54:27
   |
54 |     uri!(has_one_guarded: cookies = "hi", id = 100); //~ ERROR invalid parameters
   |                           ^^^^^^^

error: invalid parameters for `has_one_guarded` route uri
  --> $DIR/typed-uris-bad-params.rs:57:27
   |
57 |     uri!(has_one_guarded: id = 100, cookies = "hi"); //~ ERROR invalid parameters
   |                           ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32
help: unknown parameter: `cookies`
  --> $DIR/typed-uris-bad-params.rs:57:37
   |
57 |     uri!(has_one_guarded: id = 100, cookies = "hi"); //~ ERROR invalid parameters
   |                                     ^^^^^^^

error: invalid parameters for `has_two` route uri
  --> $DIR/typed-uris-bad-params.rs:60:19
   |
60 |     uri!(has_two: id = 100, id = 100, ); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32, name: String
   = help: missing parameter: `name`
help: duplicate parameter: `id`
  --> $DIR/typed-uris-bad-params.rs:60:29
   |
60 |     uri!(has_two: id = 100, id = 100, ); //~ ERROR invalid parameters
   |                             ^^

error: invalid parameters for `has_two` route uri
  --> $DIR/typed-uris-bad-params.rs:64:19
   |
64 |     uri!(has_two: name = "hi"); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32, name: String
   = help: missing parameter: `id`

error: invalid parameters for `has_two` route uri
  --> $DIR/typed-uris-bad-params.rs:67:19
   |
67 |     uri!(has_two: cookies = "hi", id = 100, id = 10, id = 10); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32, name: String
   = help: missing parameter: `name`
help: unknown parameter: `cookies`
  --> $DIR/typed-uris-bad-params.rs:67:19
   |
67 |     uri!(has_two: cookies = "hi", id = 100, id = 10, id = 10); //~ ERROR invalid parameters
   |                   ^^^^^^^
help: duplicate parameter: `id`
  --> $DIR/typed-uris-bad-params.rs:67:45
   |
67 |     uri!(has_two: cookies = "hi", id = 100, id = 10, id = 10); //~ ERROR invalid parameters
   |                                             ^^       ^^

error: invalid parameters for `has_two` route uri
  --> $DIR/typed-uris-bad-params.rs:72:19
   |
72 |     uri!(has_two: id = 100, cookies = "hi"); //~ ERROR invalid parameters
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: uri parameters are: id: i32, name: String
   = help: missing parameter: `name`
help: unknown parameter: `cookies`
  --> $DIR/typed-uris-bad-params.rs:72:29
   |
72 |     uri!(has_two: id = 100, cookies = "hi"); //~ ERROR invalid parameters
   |                             ^^^^^^^

error: path parameters cannot be ignored
  --> $DIR/typed-uris-bad-params.rs:76:26
   |
76 |     uri!(optionals: id = _, name = "bob".into());
   |                          ^

error: path parameters cannot be ignored
  --> $DIR/typed-uris-bad-params.rs:79:37
   |
79 |     uri!(optionals: id = 10, name = _);
   |                                     ^

warning: unused import: `std::fmt`
 --> $DIR/typed-uris-bad-params.rs:5:5
  |
5 | use std::fmt;
  |     ^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
